### PR TITLE
Added Group Coherent Example to develop branch for 6.1.0 release

### PR DESCRIPTION
**Pull request by mcmanuea-rti**

### CODE REVIEW

File: examples/connext_dds/group_coherent_presentation/c++11/GroupCoherentExample_publisher.cxx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    DataWriterQos writer_qos; 
+    if (use_xml_qos) {
+        // Retrieve the DataWriter QoS, from USER_QOS_PROFILES.xml.
+        writer_qos = QosProvider::Default().datawriter_qos();
+    } else {
+        // Set the DataWriter QoS programatically
+        writer_qos << Reliability::Reliable() << History::KeepAll();
+    }
+
+    // If you want to change the DataWriter's QoS programmatically rather
```
Comment (alexcamposruiz) [inline examples/connext_dds/group_coherent_presentation/c++11/GroupCoherentExample_publisher.cxx:77]: Delete this comment

File: examples/connext_dds/group_coherent_presentation/c++11/GroupCoherentExample_subscriber.cxx [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        find(subscriber, 
+             dds::sub::status::SampleState::not_read(), 
+             std::back_inserter(readers));
+
+         // Iterate through the returned readers list and take their samples
+         for (AnyDataReader reader : readers) {
+             if (reader.topic_name() == "Alarm") {
+                 dds::sub::LoanedSamples<Alarm> samples = 
+                         reader.get<Alarm>().take();
+                 for (auto& sample : samples) {
```
Comment (alexcamposruiz) [inline examples/connext_dds/group_coherent_presentation/c++11/GroupCoherentExample_subscriber.cxx:81]:
```suggestion
                 for (const auto& sample : samples) {
```

**Review by alexcamposruiz — CHANGES_REQUESTED** [review]
Looks good! Just a couple minor things.

**Review by alexcamposruiz — APPROVED** [review]

File: examples/connext_dds/group_coherent_presentation/c++11/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# (c) 2018 Copyright, Real-Time Innovations, Inc. All rights reserved.
+# No duplications, whole or partial, manual or electronic, may be made
+# without express written permission.  Any such copies, or revisions thereof,
+# must display this notice unaltered.
+# This code contains trade secrets of Real-Time Innovations, Inc.
+cmake_minimum_required(VERSION 3.7)
```
Comment (iblancasa) [inline examples/connext_dds/group_coherent_presentation/c++11/CMakeLists.txt:6]: Required version is 3.11

File: examples/connext_dds/group_coherent_presentation/c++11/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# (c) 2018 Copyright, Real-Time Innovations, Inc. All rights reserved.
```
Comment (iblancasa) [inline examples/connext_dds/group_coherent_presentation/c++11/CMakeLists.txt:1]: 2021

File: examples/connext_dds/group_coherent_presentation/c++11/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+set(CMAKE_MODULE_PATH
+    ${CMAKE_MODULE_PATH}
+    "${CMAKE_CURRENT_SOURCE_DIR}/../../../../resources/cmake"
+)
+
+# Include ConnextDdsAddExample.cmake from resources/cmake
+include(ConnextDdsAddExample)
+
+connextdds_add_example(
+	IDL "GroupCoherentExample"
```
Comment (iblancasa) [inline examples/connext_dds/group_coherent_presentation/c++11/CMakeLists.txt:17]: Indentation

File: examples/connext_dds/group_coherent_presentation/c++11/GroupCoherentExample.idl [hunk 6]
```diff
@@ -0,0 +1,27 @@
+module GroupCoherentExample
```
Comment (iblancasa) [inline examples/connext_dds/group_coherent_presentation/c++11/GroupCoherentExample.idl:1]: Missing copyright header

File: examples/connext_dds/group_coherent_presentation/c++11/application.hpp [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+            << std::endl;
+        }
+
+        return ApplicationArguments(
+                parse_result, domain_id, set_count, verbosity, use_xml_qos);
+    }
+
+}  // namespace application
+
+#endif  // APPLICATION_HPP
```
Comment (iblancasa) [inline examples/connext_dds/group_coherent_presentation/c++11/application.hpp:154]: Add newline at the end of the file

**Review by iblancasa — CHANGES_REQUESTED** [review]

**Review by iblancasa — APPROVED** [review]

**Pull request merged**